containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -240,30 +260,40 @@ class DataProductConfig:
     curated_data_bucket: str = field(default_factory=get_curated_data_bucket)
     metadata_bucket: str = field(default_factory=get_metadata_bucket)
 
+    def __post_init__(self):
+        self.latest_version = get_latest_version(self.name)
```

Comment:
the post init is a bit awkward in a dataclass but we only want 1 s3 call to find the latest version, and this was the easiest implementation I could think of